release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,240 @@
+          break;
+        case "cluster.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.CLUSTER;
+          break;
+        case "graphql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.GRAPHQL;
+          break;
+        case "sql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.SQL;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown file name: " + path);
+      }
+
+      File file = new File(path);
+      load(edition, repository, file);
+    }
+    output();
+  }
+
+  void load(Edition edition, Repository repository, File file) throws Exception {
+    Category category = null;
+    String line;
+    Matcher matcher;
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8))) {
+      while ((line = reader.readLine()) != null) {
+        if (DEBUG != null) System.err.printf("ReadLine: %s%n", line);
+        matcher = Pattern.compile("^## *(\\p{Print}+) *$").matcher(line);
+        if (matcher.matches()) {
+          if (DEBUG != null) System.err.printf("matched: %s%n", matcher.group(1));
+          if (!SECTION_SUMMARY.equalsIgnoreCase(matcher.group(1)))
+            category = Category.getByDisplayName(matcher.group(1));
+          continue;
+        }
+
+        matcher = Pattern.compile("^ *- *(\\p{Print}+)$").matcher(line);
+        if (matcher.matches()) {
+          if (category == null)
+            throw new IllegalStateException(
+                "Missing category. Release note text: " + matcher.group(1));
+          addReleaseNote(new ReleaseNote(edition, category, repository, matcher.group(1)));
+        }
+      }
+    }
+  }
+
+  private void addReleaseNote(ReleaseNote releaseNote) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap =
+        editionMap.computeIfAbsent(releaseNote.edition, k -> new EnumMap<>(Category.class));
+
+    Map<Repository, List<ReleaseNote>> repositoryMap =
+        categoryMap.computeIfAbsent(releaseNote.category, k -> new EnumMap<>(Repository.class));
+
+    List<ReleaseNote> releaseNotesList =
+        repositoryMap.computeIfAbsent(releaseNote.repository, k -> new ArrayList<>());
+
+    if (releaseNote.edition.equals(Edition.ENTERPRISE)) {
+      Matcher matcher =
+          Pattern.compile("(.*) +(\\((#[0-9]+ *)+\\))$").matcher(releaseNote.releaseNoteText);
+      if (matcher.matches()) {
+        if (DEBUG != null)
+          System.err.printf(
+              "Matched::%s::%s grp1:%s grp2:%s%n",
+              releaseNote.category, releaseNote.repository, matcher.group(1), matcher.group(2));
+        releaseNote.releaseNoteText = matcher.group(1);
+      }
+    }
+
+    releaseNotesList.add(releaseNote);
+    repositoryMap.put(releaseNote.repository, releaseNotesList);
+    categoryMap.put(releaseNote.category, repositoryMap);
+    editionMap.put(releaseNote.edition, categoryMap);
```

Comment:
I see. I'll revise it. Thank you.

Reply:
Sorry for flipping my state. I agree with you. But for now, I'd like to keep the code as is because it works without any problem for now. In addition, it is used only for creating release notes when we release the new version of Scalar products, not for production.

I will revise it according to your advice in another PR when I get a chance to revise it (e.g. Kaizen-day).